testhelper/integrationtesthelper.go

```diff
@@ -184,40 +180,36 @@ func integrationTimedLidar(
 		return nil, err
 	}
 
-	closed := false
 	injectLidar := &inject.TimedLidar{}
 	injectLidar.NameFunc = func() string { return lidar["name"] }
 	injectLidar.DataFrequencyHzFunc = func() int { return dataFrequencyHz }
 	injectLidar.TimedLidarReadingFunc = func(ctx context.Context) (s.TimedLidarReadingResponse, error) {
 		defer timeTracker.mu.Unlock()
-		/*
-			Holds the process until for all necessary IMU data has been sent to cartographer. Only applicable
-			when the IMU is present (timeTracker.NextImuTime has been defined) and is always true in the first iteration.
-			This and the manual definition of timestamps allow for consistent results.
-		*/
-		for {
-			timeTracker.mu.Lock()
-			if timeTracker.imuTime == undefinedTime {
-				time.Sleep(sensorDataIngestionWaitTime)
-				break
-			}
 
-			if i <= 1 || timeTracker.lidarTime.Sub(timeTracker.imuTime) <= 0 {
-				time.Sleep(sensorDataIngestionWaitTime)
-				break
+		if useIMU {
+			// Holds the process until IMU data has been sent to cartographer. Is always true in the first
+			// iteration. This and the manual definition of timestamps allow for consistent results.
+			for {
+				timeTracker.mu.Lock()
+				if !timeTracker.lidarTime.After(timeTracker.imuTime) {
+					break
+				}
+				timeTracker.mu.Unlock()
 			}
-			timeTracker.mu.Unlock()
+		} else {
+			timeTracker.mu.Lock()
 		}
+		time.Sleep(sensorDataIngestionWaitTime)
 
 		// Communicate that all lidar readings have been sent to cartographer or if the last IMU reading has been sent,
 		// checks if LastLidarTime has been defined. If so, simulate endOfDataSet error.
-		t.Logf("TimedLidarReading Mock i: %d, closed: %v, readingTime: %s\n", i, closed, timeTracker.lidarTime.String())
-		if i >= NumPointClouds || timeTracker.finalImuTime != undefinedTime {
+		t.Logf("TimedLidarReading Mock i: %d, closed: %v, readingTime: %s\n", i, timeTracker.lidarDone,
+			timeTracker.lidarTime.String())
+		if i >= NumPointCloudFiles || timeTracker.imuDone {
```

Comment:
[note to self] Behavior here is that if imuDone is true, then we set lidarDone to true as well and return end of dataset to prevent mode data from being sent.